cparser/Cutil.ml

```diff
@@ -533,6 +533,28 @@ let sizeof_struct env members =
       end
   in sizeof_rec 0 members
 
+(* Compute the offset of a struct member *)
+let offsetof env ty fields =
+  let rec sub acc name = function
+    | [] -> List.rev acc
+    | m::rem -> if m.fld_name = name then
+        List.rev acc
+      else
+        sub (m::acc) name rem in
+  let offset (ofs,ty) field =
+    match unroll env ty with
+    | TStruct (id,_) ->
```

Comment:
That's a good reason.  But I would prefer to see 3 cases: TStruct, TUnion, and a catch-all `_ -> assert false`